nodeup/pkg/model/nvidia.go

```diff
@@ -33,24 +31,14 @@ var _ fi.NodeupModelBuilder = &NvidiaBuilder{}
 // Build is responsible for installing packages.
 func (b *NvidiaBuilder) Build(c *fi.NodeupModelBuilderContext) error {
 	if b.InstallNvidiaRuntime() && b.Distribution.IsUbuntu() {
-		version := ""
-		if b.Distribution.Version() >= 22.04 {
-			version = "22.04"
-		} else if b.Distribution.Version() >= 20.04 {
-			version = "20.04"
-		} else {
-			version = "18.04"
-		}
 		c.AddTask(&nodetasks.AptSource{
-			Name:    "nvidia-container-runtime",
-			Keyring: "https://nvidia.github.io/nvidia-container-runtime/gpgkey",
+			Name:    "nvidia-container-toolkit",
+			Keyring: "https://nvidia.github.io/libnvidia-container/gpgkey",
 			Sources: []string{
-				fmt.Sprintf("deb https://nvidia.github.io/nvidia-container-runtime/stable/ubuntu%s/$(ARCH) /", version),
-				fmt.Sprintf("deb https://nvidia.github.io/libnvidia-container/stable/ubuntu%s/$(ARCH) /", version),
-				fmt.Sprintf("deb https://nvidia.github.io/nvidia-docker/ubuntu%s/$(ARCH) /", version),
+				"deb https://nvidia.github.io/libnvidia-container/stable/ubuntu18.04/$(ARCH) /",
```

Comment:
well we cannot hardcode it like this

Reply:
Sure, I can update this. As a matter of interest why can't it be hardcoded? If the only value is ubuntu18.04 the string manipulation is superfluous.

Reply:
I assume the ubuntu18 means 18+ in this case? So it will work with linking against libc and other libraries used in other distros. The important bit is that it works on all our supported versions.

Reply:
Based on the issue this resolves, that seems indeed the case.

Reply:
Yes, the `ubuntu18.04` packages are used for all newer Ubuntu versions. This is also the URL that is included in our repo lists that are downloaded as part of our official documentation.

Reply:
@zetaab does this explanation resolve your questions?